### PR TITLE
this commit introduces the post request, moreover, it simulates for t…

### PR DESCRIPTION
**Pull request by Hartooro**
…wo possible errors thrown.

### CODE REVIEW

File: __tests__/app.test.js [hunk 1]
```diff
@@ -129,4 +130,36 @@ describe('4 get/api/articles/:article_id', () => {
...
+        return request(app).post("/api/articles/stonks/comments").expect(400).send(comment).then((res)=>{
+            expect(res.body.msg).toBe("Bad, bad Request")
+        })
+    })
+    test('returns 400, if client tries to insert an empty object', () => {
+        const comment = {}
+        return request(app).post("/api/articles/stonks/comments").expect(400).send(comment).then((res)=>{
+            expect(res.body.msg).toBe("Bad, bad Request")
+        })
+    })
```
Comment (Rayhaan2k9) [inline __tests__/app.test.js:163]: Some more tests would be handy here, Arturo!

It's a prism of endless possibility, where a single choice can branch out into infinite realities, creating alternate worlds from the ones you know. I am the Watcher. I am your guide through these vast new realities. Follow me and ponder the question... **"What if?"**

- What if a post request is made to an article that doesn't exist? eg. `/api/articles/9999/comments`
- What if a post request is made by a user that doesn't exist in the db? eg. `{author: "Arturo", body: "Wow, have I been able to post? I'm not even registered on this site!"}`
- What if the article is valid but the comment object is empty or has missing keys? eg `{body: "I have no identity!"}`

**Review by Rayhaan2k9 — CHANGES_REQUESTED** [review]
Left a comment for you Arturo. Please add the tests I suggested and ensure they all pass. Then resubmit this PR 🙂

File: models.js [hunk 2]
```diff
@@ -64,12 +64,28 @@ exports.fetchArticleById = (id) => {
...
+                RETURNING *;
+               `, [body.body, body.author, id])
+               .then((result)=> {
+                return result.rows[0]
+               })
+            })
+        }
+
+    })
 }
```
Comment (MKD01) [inline models.js]: what happens in the event that you dont go into that if statement?

File: __tests__/app.test.js [hunk 3]
```diff
@@ -153,13 +152,32 @@ describe('4 get/api/articles/:article_id', () => {
...
+        const comment = {
+            author: "Blue power Ranger",
+            body : "Power of Water, Powers Unite! Six working together to fight evil!"
+        }
+        return request(app).post("/api/articles/4/comments").expect(404).send(comment).then((res)=>{
+    
+            expect(res.body.msg).toBe("author is nonexistant")
+        })
+    });;
   });
```
Comment (MKD01) [inline __tests__/app.test.js:191]: All of these are valid tests but what would happen if you gave a comment with a valid author but not a body? This should be a separate test.

**Review by MKD01 — CHANGES_REQUESTED** [review]
Please make these changes then send that as another PR

File: models.js (hunk 2 above)
Comment (Hartooro) [reply models.js]: @MKD01 am I right thinking that I put an else statement where I reject the promise (indicating to the user that they must insert body AND author) will also resolve the other comment you have left?

Comment (Hartooro) [reply models.js]: @MKD01 that test would try posting a comment with no body, just author

File: __tests__/app.test.js [hunk 4]
```diff
@@ -179,5 +179,13 @@ describe('4 get/api/articles/:article_id', () => {
...
             expect(res.body.msg).toBe("author is nonexistant")
         })
-    });;
+    })
+    test('returns 400 if author is valid but body is empty', () => {
+        const comment = {
+            author: "Blue power Ranger"
+        }
+        return request(app).post("/api/articles/4/comments").expect(400).send(comment).then((res)=>{
+            expect(res.body.msg).toBe("Bad, bad Request. Please make sure there's a body and an author")
```
Comment (dav3rid) [inline __tests__/app.test.js:188]: "Bad, bad request" request haha

Comment (dav3rid) [reply __tests__/app.test.js:188]: I think your `.send` and `.expect` are in the wrong order here

**Review by dav3rid — APPROVED** [review]
Approving this Arturo but be aware...
You are checking if both the article and author exist here before inserting the comment.
This is introducing additional calls to the database when you don't really need to - as the article_id and author are FOREIGN KEYs, you will find that the database will not accept the insertion if they do not exist.
This is one of the major benefits of using a relational database in the first place

**Pull request merged**